pcmanfm/mainwindow.cpp

```diff
@@ -260,6 +267,28 @@ void MainWindow::addTab(FmPath* path) {
   }
 }
 
+void MainWindow::toggleMenuBar() {
+  Settings& settings = static_cast<Application*>(qApp)->settings();
+  bool showMenuBar = !settings.showMenuBar();
+
+  if (!showMenuBar) {
+    QScopedPointer<QMessageBox> warning(
+      new QMessageBox(QMessageBox::Information,
+                      tr("Hide menu bar"),
+                      tr("This will hide the menu bar completely, use Ctrl+M to show it again."),
+                      QMessageBox::Ok | QMessageBox::Cancel));
+
+    warning->show();
+    warning->exec();
+    if (QMessageBox::Cancel == warning->result()) {
+      return;
+    }
+  }
+
+  ui.menubar->setVisible(showMenuBar);
+  settings.setShowMenuBar(showMenuBar);
```

Comment:
No, it's to update the settings so that the state of the menu bar will be remembered.